Firmware/LowLevel/src/main.cpp

```diff
@@ -583,33 +665,24 @@ void onPacketReceived(const uint8_t *buffer, size_t size) {
         if (!ROS_running) {
             // ROS is running (again (i.e. due to restart after reconfiguration))
             ROS_running = true;
-
-            // Send current LL config (and request HL config response)
-            sendConfigMessage(PACKET_ID_LL_HIGH_LEVEL_CONFIG_REQ);
         }
     } else if (buffer[0] == PACKET_ID_LL_HIGH_LEVEL_STATE && size == sizeof(struct ll_high_level_state)) {
         // copy the state
         last_high_level_state = *((struct ll_high_level_state *) buffer);
-    }
-    else if ((buffer[0] == PACKET_ID_LL_HIGH_LEVEL_CONFIG_REQ || buffer[0] == PACKET_ID_LL_HIGH_LEVEL_CONFIG_RSP) && size == sizeof(struct ll_high_level_config))
-    {
-        // Read and handle received config
-        struct ll_high_level_config *pkt = (struct ll_high_level_config *)buffer;
-        if (pkt->comms_version <= LL_HIGH_LEVEL_CONFIG_MAX_COMMS_VERSION)
-            comms_version = pkt->comms_version;
-        else
-            comms_version = LL_HIGH_LEVEL_CONFIG_MAX_COMMS_VERSION;
-        config_bitmask = pkt->config_bitmask; // Take over as sent. HL is leading (for now)
-        // FIXME: Assign volume & language if not already stored in flash-config
+    } else if (buffer[0] == PACKET_ID_LL_HIGH_LEVEL_CONFIG_REQ || buffer[0] == PACKET_ID_LL_HIGH_LEVEL_CONFIG_RSP) {
+        applyConfig(buffer + 1, size - 1);  // Skip type
+
+        // Store in flash
+        saveConfigToFlash();
```

Comment:
This will save the effective config, right? If we later change any defaults, it will still load the old values from flash. I think we need to save the packet from ROS instead. We can even store it in full length and keep the CRC, because applyConfig() will be used when loading it.

Reply:
Yes, think you're right! Good hint!!

Reply:
Really clever hint!! :+1: 
Committed, not yet tested, but think should work :innocent: